sdks/java/io/xml/src/main/java/org/apache/beam/sdk/io/xml/XmlSource.java

```diff
@@ -320,7 +330,12 @@ private long getFirstOccurenceOfRecordElement(
             recordStartBytesMatched = true;
           }
         }
-        buf.clear();
+        if (bufSizeChanged) {
+          buf = ByteBuffer.allocate(BUF_SIZE);
```

Comment:
Add a comment.
"We have to reset the size of the buffer to 'BUF_SIZE' to prevent it from infinitely increasing."

Reply:
done